src/IceRpc.Protobuf/MessageParserExtensions.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) ZeroC, Inc.
+
+using Google.Protobuf;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.IO.Pipelines;
+
+namespace IceRpc.Protobuf;
+
+/// <summary>Provides an extension method for <see cref="MessageParser{T}" />.</summary>
+public static class MessageParserExtensions
+{
+    /// <summary>Decodes a Protobuf length prefixed message from a <see cref="PipeReader" />.</summary>
```

Comment:
That's not quite correct. Here, you're decoding a Protobuf message _from_ a stream of bytes (the pipe reader). And the format of these bytes is a length-prefixed message.

Reply:
Maybe it would be cleaner to make it a PipeReader extension method `DecodeProtobufMessageAsync(...)`?

Reply:
Yes indeed